rust/parser/mod.rs

```diff
@@ -33,19 +33,28 @@ use antlr_rust::{
 use chrono::{NaiveDate, NaiveDateTime};
 use std::rc::Rc;
 
-use crate::common::{
-    date_time,
-    error::{ErrorMessage, ILLEGAL_GRAMMAR},
-    string::*,
-    token::Predicate,
-};
-use typeql_grammar::typeqlrustparser::*;
-
 use crate::{
-    common::token::{Aggregate, ValueType},
-    pattern::*,
-    query::*,
+    common::{
+        date_time,
+        error::{ErrorMessage, ILLEGAL_GRAMMAR},
+        string::{unescape_regex, unquote},
+        token,
+    },
+    pattern::{
+        ConceptVariable, ConceptVariableBuilder, Conjunction, Disjunction, HasConstraint,
+        IsKeyAttribute, IsaConstraint, Label, Negation, OwnsConstraint, Pattern, PlaysConstraint,
+        RelatesConstraint, RelationConstraint, RolePlayerConstraint, RuleDeclaration,
+        RuleDefinition, SubConstraint, ThingConstrainable, ThingVariable, ThingVariableBuilder,
+        Type, TypeConstrainable, TypeVariable, TypeVariableBuilder, UnboundVariable, Value,
+        ValueConstraint, Variable,
+    },
+    query::{
+        sorting, AggregateQueryBuilder, Query, Sorting, TypeQLDefine, TypeQLDelete, TypeQLInsert,
+        TypeQLMatch, TypeQLMatchAggregate, TypeQLMatchGroup, TypeQLMatchGroupAggregate,
+        TypeQLUndefine, TypeQLUpdate,
+    },
 };
+use typeql_grammar::typeqlrustparser::*;
```

Comment:
Not expanding this star import because I don't think we should care about the exact types we're importing here -- it's just all the contexts and terminal nodes.

Reply:
The reason we avoid star imports in general (in all programming languages) is so that, given any symbol in the file, it's easy to look at the import list and see where the symbol comes from.

Leaving this star import in would break that convention.

Reply:
The idea here is that we want to import all the contexts from the generated grammar, because those are the types we intend to visit. Listing all fifty-odd types seems needlessly noisy.